nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,177 @@
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = session.createDocumentModel("/", "domain", "Domain");
+        domain = session.createDocument(domain);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToMainDocument() {
+        // create a File document and
+        DocumentModel documentModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification
+        addSubscriptions(documentModel);
+
+        // fire the dummy event
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), documentModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(documentModel, DOCUMENT_NAME);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToParentDocument() {
+        // create a simple hierarchy (domain <- DOCUMENT_NAME <- ANOTHER_DOCUMENT_NAME)
+        DocumentModel parentDocModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification on the parent document
+        addSubscriptions(parentDocModel);
+
+        // create a child document
+        DocumentModel mainDocModel = createDocument(parentDocModel, ANOTHER_DOCUMENT_NAME);
+
+        // fire the dummy event on the child document
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), mainDocModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(mainDocModel, ANOTHER_DOCUMENT_NAME);
+    }
+
+    protected void checkMailContent(final DocumentModel documentModel, String documentName) {
+        assertEquals(1, emailsResult.getMails().size());
+        SmtpMailServerFeature.MailMessage mailMessage = emailsResult.getMails().get(0);
+        assertEquals(String.format("[Dummy]Notification on the document '%s'", documentName), mailMessage.getSubject());
+
+        // check the mail content (see templates/dummyNotificationToSendMail.ftl)
+        String mailContent = getMailContent(mailMessage);
+        // description
+        assertTrue(mailContent.contains(String.format("The document %s is now available.", documentName)));
+        // author
+        assertTrue(mailContent.contains(session.getPrincipal().getName()));
+        // location
+        assertTrue(mailContent.contains(documentModel.getPathAsString()));
+        // version
+        assertTrue(mailContent.contains(documentModel.getVersionLabel()));
+        // state
+        assertTrue(mailContent.contains(documentModel.getCurrentLifeCycleState()));
+        // link to the document
+        assertTrue(mailContent.contains(String.format("Consult the document %s", documentName)));
```

Comment:
Why not match the full mail content, maybe with a regexp for the parts that are too random to predict (if any)?

Reply:
Even better, you could have a file with the expected content, using variables to be expended in the test.
Then you would only need to compare the received and expected email strings.

See what is done for WOPI to check some JSON responses:

https://github.com/nuxeo/nuxeo/blob/master/addons/nuxeo-wopi/nuxeo-wopi/src/test/java/org/nuxeo/wopi/jaxrs/TestFilesEndpoint.java#L1235

https://github.com/nuxeo/nuxeo/blob/master/addons/nuxeo-wopi/nuxeo-wopi/src/test/java/org/nuxeo/wopi/JSONHelper.java#L35

Reply:
yep i did the same things on comments / annotation notification,  please see https://github.com/nuxeo/nuxeo/pull/3705/files#r372356410 if it suits you.